extensions/applicationinsights-react-js/src/ReactPlugin.ts

```diff
@@ -69,6 +71,15 @@ export default class ReactPlugin extends BaseTelemetryPlugin {
         }
     }
 
+    trackEvent(event: IEventTelemetry, customProperties?: ICustomProperties) {
+        if (this._analyticsPlugin) {
+            this._analyticsPlugin.trackEvent(event, customProperties);
+        } else {
+            this._logger.throwInternal(
```

Comment:
So this then becomes

this.diagLog().throwInternal(...)

Reply:
I'm not seeing any other blocking issues apart from the intro of the _logger.
For references the diaLog() takes care of the this getting called before initialization() as well.